forge-gui/src/main/java/forge/gamemodes/net/NetConnectUtil.java

```
@@ -122,7 +122,22 @@ public static void copyHostedServerUrl() {
 
     public static ChatMessage join(final String url, final IOnlineLobby onlineLobby, final IOnlineChatInterface chatInterface) {
         final IGuiGame gui = GuiBase.getInterface().getNewGuiGame();
-        final FGameClient client = new FGameClient(FModel.getPreferences().getPref(FPref.PLAYER_NAME), "0", gui);
+        String hostname = url;
+        int port = FModel.getNetPreferences().getPrefInt(ForgeNetPreferences.FNetPref.NET_PORT);
+
+        //see if port specified in URL
+        int urlSeparatorIndex = url.indexOf(':');
```

I know this isn't your code, but I think this can be written cleaner.

---

int port = FModel.getNetPreferences().getPrefInt(ForgeNetPreferences.FNetPref.NET_PORT);

If this is the line you are refering too, i can shove it off into a function getServerConfigPort(), that should make following the code easier. If your refering to the url serperator code beneath i can look into making that better

---

I just meant this whole block of code. It's basically trying to split by colon and do somethings based off the length, but its doing it awkwardly and swallowing an exception in the meantime.

---

ok, i replaced it with a URI parcer, looks much nicer and should hopefully be much easier to follow along with